pkgs/applications/misc/input-leap/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib, fetchFromGitHub, cmake, curl, xorg, avahi, qtbase, qttools, libei ? null, libportal,
+  mkDerivation, pkg-config, openssl, wrapGAppsHook,
+  avahiWithLibdnssdCompat ? avahi.override { withLibdnssdCompat = true; },
+  fetchpatch
+}:
+
+mkDerivation rec {
+  pname = "input-leap";
+  version = "unstable-2023-05-24";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "5e2f37bf9ec17627ae33558d99f90b7608ace422";
+    hash = "sha256-55RqdRu/Hi2OTiLjAFJ6Gdgg9iO5NIIJCsOkUQjR9hk=";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [
+    curl qtbase avahiWithLibdnssdCompat
+  ] ++ (with xorg; [
+    libX11 libXext libXtst libXinerama libXrandr libXdmcp libICE libSM
```

Comment:
```suggestion
```

Declare xorg dependencies explicitly. There is no use for `with xorg`.

Reply:
I'm a bit confused, are you suggesting that e.g. `libXext` should be directly in the arguments and we should unpack all these libraries from `xorg` in `all-packages.nix`?

Reply:
I interpreted this as using something like `++ [ xorg.libX11 xorg.libXext (etc) ]` - but I can't see much benefit over the other.

Reply:
To be more precise,

`++ [ libX11 libXext (etc) ]`

Reply:
But where would those things come from? They live in `xorg`, so they aren't visible unprefixed without the `with xorg`.

Reply:
They **are** visible without `xorg.` prefix. Just put them on the input attribute set.